pkg/apis/extensions/types.go

```diff
@@ -437,6 +449,22 @@ type RollingUpdateDaemonSet struct {
 	MaxUnavailable intstr.IntOrString
 }
 
+// Spec to control the desired behavior of a daemon set surging rolling update.
+type SurgingRollingUpdateDaemonSet struct {
```

Comment:
deployments put both MaxUnavailable and MaxSurge in the rolling update strategy parameters. is there a reason not to do the same here?